bindings/java/src/main/java/org/openucx/jucx/ucp/UcpConstants.java

```diff
@@ -155,5 +155,31 @@
     public static long UCP_AM_SEND_FLAG_EAGER;
     public static long UCP_AM_SEND_FLAG_RNDV;
 
+    /**
+     * Flags for a UCP Active Message callback.
+     */
+
+    /**
+     * Indicates that the entire message will be handled in one callback. With this
+     * option, message ordering is not guaranteed (i.e. receive callbacks may be
+     * invoked in a different order than messages were sent).
+     * If this flag is not set, the data callback may be invoked several times for
+     * the same message (if, for example, it was split into several fragments by
+     * the transport layer). It is guaranteed that the first data callback for a
+     * particular message is invoked for the first fragment. The ordering of first
+     * message fragments is guaranteed (i.e. receive callbacks will be called
+     * in the order the messages were sent). The order of other fragments is not
+     * guaranteed. User header is passed with the first fragment only.
+     */
+    public static long UCP_AM_FLAG_WHOLE_MSG;
+
+    /**
+     * Guarantees that the specified {@link UcpAmRecvCallback#onReceive} callback,
+     * will always be called with {@link UcpConstants#UCP_AM_RECV_ATTR_FLAG_DATA} flag set,
+     * so the data will be accessible outside the callback, until {@link UcpWorker#amDataRelease}
```

Comment:
what does it mean for the Java API?

Reply:
That it's safe to return `IN_PROGRESS` from amData callback.

Reply:
w/o it, it's possible that `amData.isDataValid = true`, but user can't return `IN_PROGRESS`

Reply:
so i think need to update the documentation, since AFAIU `UCP_AM_RECV_ATTR_FLAG_DATA` is not exposed directly in Java API. Can we describe it by referring to `isDataValid`?

Reply:
exposed flags explicitly. Because with this PR it's possible to not set `PERSISTENT_DATA` flag, so `isDataValid` would be true, but it could not be used for persistance

Reply:
TODO:  add mutifragment support to have 1:1 functionality with ucp AM.  I can do this in this or separate PR

Reply:
it is not fully completed yet, can be slightly updated in the short term